### PR TITLE
owner cleanliness test changes and tweaks

### PR DESCRIPTION
**Pull request by czue**
mostly tests + improving behavior when the sync mode changes between syncs.

@snopoke cc @TylerSheffels

### CODE REVIEW

**Comment by snopoke** [comment]
so what's the plan for migrating users from old to new?

**Comment by czue** [comment]
so I was being lazy here. the phone will 412 and then there will be a new initial sync and then things will work as expected.

the information in the new token format is a subset of what's in the old one, so I think we can just migrate the sync token and pick up on a steady-state sync with no 412s, but I didn't have time to write/test that properly yet.

i think that will be a requirement for TDH so planning on doing it whenever I have a bit of time.

also, I think we'll want to use a similar approach to timezones when rolling this out globally, so I want to discuss with @dannyroberts on the best approach for that and whether we can share anything.

**Comment by snopoke** [comment]
Sounds good.

**Comment by dannyroberts** [comment]
@czue I think my initial instinct is there's little in terms of direct code that's sharable but that the approach can be very similar. I think I'd be wary of writing a "by-domain data migration tracking framework" without having done it a few times and knowing we're solving the right problem.

**Comment by czue** [comment]
cool, thanks. I'll take a look at what you've done and ping you if I have
any questions.

File: corehq/ex-submodules/casexml/apps/phone/restore.py [hunk 1]
```diff
@@ -331,13 +331,19 @@ def __init__(self, project, user, params):
...
         check_version(self.params.version)
-        if self.last_sync_log and self.params.state_hash:
-            parsed_hash = CaseStateHash.parse(self.params.state_hash)
-            computed_hash = self.last_sync_log.get_state_hash()
-            if computed_hash != parsed_hash:
-                raise BadStateException(expected=computed_hash,
-                                        actual=parsed_hash,
-                                        case_ids=self.last_sync_log.get_footprint_of_cases_on_phone())
+        if self.last_sync_log:
+            if type(self.last_sync_log) != self.sync_log_class:
```
Comment (millerdev) [inline corehq/ex-submodules/casexml/apps/phone/restore.py]: Testing type equality is generally discouraged, and it looks like the type hierarchy does not preclude using `isinstance()` here. Was there a reason you decided against it?

Comment (czue) [inline corehq/ex-submodules/casexml/apps/phone/restore.py]: I think at the time of writing I forgot that `SimplifiedSyncLog` didn't extend `SyncLog`. will fix

Comment (czue) [inline corehq/ex-submodules/casexml/apps/phone/restore.py]: https://github.com/dimagi/commcare-hq/pull/7037

**Pull request merged**